eZ/Publish/Core/MVC/Symfony/Routing/UrlWildcardRouter.php

```diff
@@ -0,0 +1,210 @@
+            }
+        }
+
+        if (isset($this->logger)) {
+            $this->logger->info("UrlWildcard matched. Destination URL: {$path}");
+        }
+
+        // redirect if URLWildcard is configured as forward
+        if ($redirect) {
+            $params = [
+                '_controller' => 'Symfony\Bundle\FrameworkBundle\Controller\RedirectController::urlRedirectAction',
+                'path' => $path,
+                'keepRequestMethod' => true,
+            ];
+
+            return $params;
+        }
+
+        // set translated path for the next router
+        $request->attributes->set('semanticPathinfo', $path);
+        $request->attributes->set('needsRedirect', false);
+
+        // and throw Exception to pass processing to the next router
+        throw new ResourceNotFoundException();
+    }
+
+    public function getRouteCollection(): RouteCollection
+    {
+        return new RouteCollection();
+    }
+
+    public function generate($name, $parameters = array(), $referenceType = self::ABSOLUTE_PATH)
+    {
+        throw new RouteNotFoundException('Could not match route');
+    }
+
+    public function setContext(RequestContext $context): void
+    {
+        $this->requestContext = $context;
+        $this->generator->setRequestContext($context);
+    }
+
+    public function getContext()
+    {
+        $this->requestContext;
+    }
+
+    /**
+     * Not supported. Please use matchRequest() instead.
+     *
+     * @param $pathinfo
+     *
+     * @throws \RuntimeException
+     */
+    public function match($pathinfo)
+    {
+        throw new \RuntimeException("The UrlWildcardRouter doesn't support the match() method. Please use matchRequest() instead.");
+    }
+
+    /**
+     * Whether the router supports the thing in $name to generate a route.
+     *
+     * This check does not need to look if the specific instance can be
+     * resolved to a route, only whether the router can generate routes from
+     * objects of this class.
+     *
+     * @param mixed $name The route name or route object
+     *
+     * @return bool
+     */
+    public function supports($name): bool
+    {
+        return $name === self::URL_ALIAS_ROUTE_NAME;
+    }
+
+    /**
+     * @see Symfony\Cmf\Component\Routing\VersatileGeneratorInterface::getRouteDebugMessage()
+     */
+    public function getRouteDebugMessage($name, array $parameters = array())
+    {
+        if ($name instanceof RouteObjectInterface) {
+            return 'Route with key ' . $name->getRouteKey();
+        }
+
+        if ($name instanceof SymfonyRoute) {
+            return 'Route with pattern ' . $name->getPath();
+        }
+
+        return $name;
+    }
+
+    /**
+     * Returns the UrlWildcard object.
+     *
+     * @param string $requestedPath
+     *
+     * @return \eZ\Publish\API\Repository\Values\Content\URLWildcard
+     *
+     * @throws NotFoundException
+     */
+    protected function getUrlWildcard($requestedPath): APIUrlWildcard
```

Comment:
these methods does not seem to add any value besides making the code harder to read, or?

Reply:
Indeed, removed them.